paylike.php

```diff
@@ -109,225 +151,160 @@ public function renderForm()
+				else
+					$this->context->controller->confirmations[] = $this->l('Transction captured successfully.');
+
+			}
+		}
+	}
 
+	public function hookPayment($params)
+	{
 		//ensure paylike key is set
 		if (!Configuration::get('PAYLIKE_API_KEY') || !Configuration::get('PAYLIKE_APP_KEY'))
 			return false;
 
-		global $smarty,$cookie;
-
 		$products = $params['cart']->getProducts();
-
+		$customer = new Customer((int)$params['cart']->id_customer);
 		$productnames = array();
-		$paylikeproductsarray = '';
-
-		foreach ($products as $product) {
+		$paylikeproductsarray = array();
+		$customer_data = array();
+		$other_data = array();
+		$customer_data[] = array(
+			$this->l('First Name') => $customer->firstname,
+			$this->l('Last Name') => $customer->lastname,
+			$this->l('Email') => $customer->email
+			);
+		$other_data[] = array(
+			$this->l('Shop Name') => $this->context->shop->name,
+			$this->l('Prestashop version') => _PS_VERSION_,
+			$this->l('Module version') => $this->version,
+			);
+		foreach ($products as $product)
+		{
 			$productnames[] = $product['name'];
-			$paylikeproductsarray .= '{SKU:"'.$product["id_product"].'", quantity:'.$product["cart_quantity"].'},';
+			$paylikeproductsarray[] = array(
+				$this->l('Product Name') => $product['name'],
+				$this->l('SKU') => $product['id_product'],
+				$this->l('Quantity') => $product['cart_quantity']
+			);
 		}
 
-		$description = implode(',',$productnames);
-
-
-		$amount = $params["cart"]->getOrderTotal() * 100;//pad amounts with 100 to handle paylike's decimals
-
-
-		$currency = Db::getInstance()->getValue('
-			SELECT `iso_code`
-			FROM `'._DB_PREFIX_.'currency`
-			WHERE `id_currency` = '.$params["cart"]->id_currency);
-
-
-	return '
-	<script src="https://sdk.paylike.io/2.js"></script>
-	<script>
-
-			var paylike = Paylike("'.Configuration::get("PAYLIKE_API_KEY").'");
-
-			function pay(){
-				paylike.popup({
-					title: "'.$this->context->shop->name.'",
-					description: "'.$description.'",
-					currency: "'.$currency.'",
-					amount: '.$amount.',
-					descriptor: "Payment to '.$this->context->shop->name.'",
-
-						custom: {
-							products: [
-								'.$paylikeproductsarray.'
-							],
-						},
-
-				}, function( err , r){
-					if (err)
-					{
-						return console.warn(err);
-					}
-
-					location.href = "'.(Configuration::get("PS_SSL_ENABLED") ? "https" : "http").'://'.$_SERVER['HTTP_HOST'].__PS_BASE_URI__.'modules/paylike/paymentReturn.php?transactionid="+r.transaction.id;
-				});
-			}
-		</script>'.$this->display(__FILE__, 'payment.tpl');
+		$description = implode(',', $productnames);
+		$amount = $params['cart']->getOrderTotal() * 100;//paid amounts with 100 to handle paylike's decimals
+
+		$currency = new Currency((int)$params['cart']->id_currency);
+		$this->context->smarty->assign(array(
+			'PAYLIKE_APP_KEY'		=> Configuration::get('PAYLIKE_API_KEY'),
```

Comment:
No, it isn't naming typo, however Prestashop standard recommends to use capitalize Configuration values (not mandatory). I will use public and secret as you suggested.